documentation/src/docs/security/securing-distributed-mode-cluster.md

```diff
@@ -0,0 +1,420 @@
+   * [Conclusion](#conclusion)
+
+## Introduction
+
+In the [distributed mode](../deployment/deployment.md#pravega-modes) of running a Pravega cluster, each service runs
+separately on one or more processes, usually spread across multiple machines. The deployment options of this mode
+include:
+
+1. A manual deployment in hardware or virtual machines
+2. Containerized deployments of these types:
+    * A Kubernetes native application deployed using the Pravega Operator
+    * A Docker Compose application deployment
+    * A Docker Swarm based distributed deployment
+
+Regardless of the deployment option used, setting up Transport Layer Security (SSL/TLS) and client auth (short for
+authentication and authorization) are important steps towards a secure Pravega deployment.
+
+TLS encrypts client-server and internal communication. It also enables clients to authenticate the services running on the server nodes. Client auth enables the services to authenticate and authorize the clients. Pravega strongly recommends enabling both TLS and auth, for production clusters.
+
+Setting up security - especially TLS - in a large cluster can be daunting at first. To make it easier, this document
+provides step-by-step instructions on how to enable and configure security manually.
+
+Depending on the deployment option used and your environment, you might need to modify the steps and commands to
+suit your specific needs and policies.
+
+## Setting Up SSL/TLS
+
+There are broadly two ways of using TLS for client-server communications:
+
+1. Setup Pravega to handle TLS directly. In this case, end-to-end traffic is encrypted.
+2. Terminate TLS outside of Pravega, in an infrastructure component such as a reverse proxy or a load balancer. Traffic is encrypted until the terminating point and is in plaintext from there to Pravega.
+
+Depending on the deployment option used, it might be easier to use one or the other approach. For example, if you
+are deploying a Pravega cluster in Kubernetes, you might find approach 2 simpler and easier to manage. In a cluster deployed manually on hardware machines, it might be more convenient to use approach 1 in many cases.
+The specifics of enabling TLS will also differ depending on the deployment option used.
+
+Here, we describe the steps applicable for approach 1 in manual deployments. For approach 2, refer to the platform vendor's documentation.
+
+At a high level, setting up TLS can be divided into three distinct stages:
+
+1. Setting up a Certificate Authority (CA)
+2. Obtaining server certificates and keys
+3. Enabling TLS and Deploying certificates
+
+They are described in detail in the following sub-sections.
+
+**Before you Begin**
+
+As the steps in this section use either OpenSSL or Java Keytool, install
+OpenSSL and Java Development Kit (JDK) on the hosts that will be used to  generate
+TLS certificates, keys, keystores are truststores.
+
+NOTE:
```

Comment:
**Note:** The examples shown in this section use command line arguments to pass all inputs to the command. To pass
sensitive command arguments via prompts instead, just exclude the corresponding option. For example,

Reply:
Changed the formatting for "Note". There are two points, so they come one after another.